jtuples/src/main/java/org/jtuples/Tuples.java

```diff
@@ -0,0 +1,219 @@
+     * @param <C> the type of the third element of the triple
+     * @param first the first element of the triple
+     * @param second the second element of the triple
+     * @param third the third element of the triple
+     * @return a triple with the given arguments
+     */
+    public static <A, B, C> Triple<A, B, C> with(A first, B second, C third) {
+        return new Triple<>(first, second, third);
+    }
+    
+    /**
+     * Creates a quadruple with the given arguments.
+     * @param <A> the type of the first element of the quadruple
+     * @param <B> the type of the second element of the quadruple
+     * @param <C> the type of the third element of the quadruple
+     * @param <D> the type of the fourth element of the quadruple
+     * @param first the first element of the quadruple
+     * @param second the second element of the quadruple
+     * @param third the third element of the quadruple
+     * @param fourth the fourth element of the quadruple
+     * @return a quadruple with the given arguments
+     */
+    public static <A, B, C, D> Quadruple<A, B, C, D> with(A first, B second,
+            C third, D fourth) {
+        return new Quadruple<>(first, second, third, fourth);
+    }
+    
+    /**
+     * Creates a quintuple with the given arguments.
+     * @param <A> the type of the first element of the quintuple
+     * @param <B> the type of the second element of the quintuple
+     * @param <C> the type of the third element of the quintuple
+     * @param <D> the type of the fourth element of the quintuple
+     * @param <E> the type of the fifth element of the quintuple
+     * @param first the first element of the quintuple
+     * @param second the second element of the quintuple
+     * @param third the third element of the quintuple
+     * @param fourth the fourth element of the quintuple
+     * @param fifth the fifth element of the quintuple
+     * @return a quintuple with the given arguments
+     */
+    public static <A, B, C, D, E> Quintuple<A, B, C, D, E> with(A first,
+            B second, C third, D fourth, E fifth) {
+        return new Quintuple<>(first, second, third, fourth, fifth);
+    }
+    
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatFirst(Pair<Pair<A, B>, C> pair) {
+        return new Triple<>(
+                pair.first().first(),
+                pair.first().second(),
+                pair.second()
+        );
+    }
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatSecond(Pair<A, Pair<B, C>> pair) {
+        return new Triple<>(
+                pair.first(),
+                pair.second().first(),
+                pair.second().second()
+        );
+    }
+    
+    
+    public static <A, B, C>
+            Pair<Pair<A, B>, C> associateFirst(Pair<A, Pair<B, C>> pair) {
+        return new Pair<>(
+                new Pair<>(pair.first(), pair.second().first()),
+                pair.second().second()
+        );
+    }
```

Comment:
/**
     * Associates a nested tuple on the first position, resulting in a tuple containing all the previous elements,
     * preserving order, but moving the position of the nesting.
     * That is, for a pair {@code (a, (b, c))} the result is the pair {@code ((a, b), c)}.
     */